webui/react/src/components/JupyterLabModal.tsx

```diff
@@ -376,13 +318,19 @@ const JupyterLabForm: React.FC<{
   currentWorkspace?: Workspace;
   defaults: JupyterLabOptions;
   form: FormInstance<JupyterLabOptions>;
+  setButtonDisabled: (buttonDisabled: boolean) => void;
   workspaces: Workspace[];
-}> = ({ currentWorkspace, form, defaults, workspaces }) => {
+}> = ({ currentWorkspace, form, defaults, setButtonDisabled, workspaces }) => {
   const [templates, setTemplates] = useState<Template[]>([]);
 
   const resourcePools = Loadable.getOrElse([], useObservable(useClusterStore().resourcePools));
 
   const selectedPoolName = Form.useWatch('pool', form);
+  const workspace = Form.useWatch('workspaceId', form);
+
+  useEffect(() => {
+    setButtonDisabled(!workspace);
```

Comment:
I thought the modal was supposed to automatically disable the button if a contained form was in an invalid state?

Reply:
not automatically since the modal doesnt have access to the state of child component, but it does have a prop for submit being enabled.

we could have people provide a form ref in the modal props, but that's more opaque and they're still providing an extra (presumably optional) prop.

that said, we can probably do `const buttonDisabled = !workspace` without the additional `useState` here